diff --git a/node_modules/@legendapp/list/index.d.ts b/node_modules/@legendapp/list/index.d.ts
index e939fc2..6371be7 100644
--- a/node_modules/@legendapp/list/index.d.ts
+++ b/node_modules/@legendapp/list/index.d.ts
@@ -506,7 +506,7 @@ declare const LegendList: <T>(props: Omit<react_native.ScrollViewProps, "content
     viewabilityConfig?: ViewabilityConfig;
     viewabilityConfigCallbackPairs?: ViewabilityConfigCallbackPairs | undefined;
     waitForInitialLayout?: boolean;
-} & React$1.RefAttributes<LegendListRef>) => React$1.ReactNode;
+} & React$1.RefAttributes<LegendListRef>) => JSX.Element;
 
 declare function useViewability(configId: string, callback: ViewabilityCallback): void;
 declare function useViewabilityAmount(callback: ViewabilityAmountCallback): void;
