diff --git a/node_modules/@react-navigation/drawer/lib/commonjs/views/DrawerView.js b/node_modules/@react-navigation/drawer/lib/commonjs/views/DrawerView.js
index 0563a7c..6315acb 100644
--- a/node_modules/@react-navigation/drawer/lib/commonjs/views/DrawerView.js
+++ b/node_modules/@react-navigation/drawer/lib/commonjs/views/DrawerView.js
@@ -32,6 +32,7 @@ function DrawerViewBase({
   descriptors,
   defaultStatus,
   drawerContent = renderDrawerContentDefault,
+  drawerStyle,
   detachInactiveScreens = _reactNative.Platform.OS === 'web' || _reactNative.Platform.OS === 'android' || _reactNative.Platform.OS === 'ios'
 }) {
   const {
@@ -42,7 +43,6 @@ function DrawerViewBase({
     drawerHideStatusBarOnOpen,
     drawerPosition = direction === 'rtl' ? 'right' : 'left',
     drawerStatusBarAnimation,
-    drawerStyle,
     drawerType = _reactNative.Platform.select({
       ios: 'slide',
       default: 'front'
