diff --git a/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java b/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java
index bdd08f1..251a716 100644
--- a/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java
+++ b/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManager.java
@@ -240,4 +240,14 @@ public class ReactSliderManager extends SimpleViewManager<ReactSlider> implement
     public Map<String, Object> getExportedCustomDirectEventTypeConstants() {
         return ReactSliderManagerImpl.getExportedCustomDirectEventTypeConstants();
     }
+    
+    @Override
+    public void setSliderThickness(ReactSlider view, double value) {
+      ReactSliderManagerImpl.setSliderThickness(view, value);
+    }
+
+    @Override
+    public void setSliderCornerRoundness(ReactSlider view, double value) {
+      ReactSliderManagerImpl.setSliderCornerRoundness(view, value);
+    }
 }
diff --git a/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManagerImpl.java b/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManagerImpl.java
index 4d76b98..2e29f82 100644
--- a/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManagerImpl.java
+++ b/node_modules/@react-native-community/slider/android/src/main/java/com/reactnativecommunity/slider/ReactSliderManagerImpl.java
@@ -1,11 +1,14 @@
 package com.reactnativecommunity.slider;
 
+
 import android.graphics.PorterDuff;
 import android.graphics.PorterDuffColorFilter;
 import android.graphics.drawable.Drawable;
 import android.graphics.drawable.LayerDrawable;
 import android.os.Build;
 
+import static com.facebook.drawee.drawable.RoundedCornersDrawable.Type.CLIPPING;
+import com.facebook.drawee.drawable.RoundedCornersDrawable;
 import com.facebook.react.bridge.ReadableArray;
 import com.facebook.react.bridge.ReadableMap;
 import com.facebook.react.common.MapBuilder;
@@ -144,4 +147,22 @@ public class ReactSliderManagerImpl {
                 ReactSlidingCompleteEvent.EVENT_NAME, MapBuilder.of("registrationName", ReactSlidingCompleteEvent.EVENT_NAME)
         );
     }
+    
+    public static void setSliderThickness(ReactSlider view, double value) {
+        LayerDrawable drawable = (LayerDrawable) view.getProgressDrawable().getCurrent();
+        for (int i = 0; i < drawable.getNumberOfLayers(); i ++ ) {
+            // 0 is max/background progress; 1 is ???; 2 is min/current progress
+            drawable.setLayerHeight(i, (int) value);
+        }
+    }
+
+    public static void setSliderCornerRoundness(ReactSlider view, double value) {
+        LayerDrawable drawable = (LayerDrawable) view.getProgressDrawable().getCurrent();
+        for (int i = 0; i < drawable.getNumberOfLayers(); i ++ ) {
+            RoundedCornersDrawable newDrawable = new RoundedCornersDrawable(drawable.getDrawable(i));
+            newDrawable.setRadius((float) value);
+            newDrawable.setType(CLIPPING);
+            drawable.setDrawable(i, newDrawable);
+        }
+    }
 }
diff --git a/node_modules/@react-native-community/slider/package.json b/node_modules/@react-native-community/slider/package.json
index 6317968..1935a2f 100644
--- a/node_modules/@react-native-community/slider/package.json
+++ b/node_modules/@react-native-community/slider/package.json
@@ -8,7 +8,7 @@
   "publishConfig": {
     "access": "public"
   },
-  "main": "dist/Slider.js",
+  "main": "src/Slider.tsx",
   "types": "typings/index.d.ts",
   "keywords": [
     "react-native",
diff --git a/node_modules/@react-native-community/slider/src/RNCSliderNativeComponent.ts b/node_modules/@react-native-community/slider/src/RNCSliderNativeComponent.ts
index 62f97ab..ded9951 100644
--- a/node_modules/@react-native-community/slider/src/RNCSliderNativeComponent.ts
+++ b/node_modules/@react-native-community/slider/src/RNCSliderNativeComponent.ts
@@ -39,6 +39,8 @@ export interface NativeProps extends ViewProps {
   value?: Float;
   lowerLimit?: Float;
   upperLimit?: Float;
+  sliderThickness?: Double;
+  sliderCornerRoundness?: Double;
 }
 
 export default codegenNativeComponent<NativeProps>('RNCSlider', {
diff --git a/node_modules/@react-native-community/slider/src/Slider.tsx b/node_modules/@react-native-community/slider/src/Slider.tsx
index c452ef9..5cc9d10 100644
--- a/node_modules/@react-native-community/slider/src/Slider.tsx
+++ b/node_modules/@react-native-community/slider/src/Slider.tsx
@@ -38,6 +38,18 @@ type WindowsProps = Readonly<{
   vertical?: boolean;
 }>;
 
+type AndroidProps = Readonly<{
+  /**
+   * modifies slider thickness. Android only.
+   */
+  sliderThickness?: number;
+  /**
+   * modifies slider corner roundness. Android only.
+   */
+  sliderCornerRoundness?: number;
+}>;
+
+
 type IOSProps = Readonly<{
   /**
    * Assigns a single image for the track. Only static images are supported.
@@ -66,6 +78,7 @@ type IOSProps = Readonly<{
 
 type Props = ViewProps &
   IOSProps &
+  AndroidProps &
   WindowsProps &
   Readonly<{
     /**
diff --git a/node_modules/@react-native-community/slider/typings/index.d.ts b/node_modules/@react-native-community/slider/typings/index.d.ts
index 6903ca7..e4d2db8 100644
--- a/node_modules/@react-native-community/slider/typings/index.d.ts
+++ b/node_modules/@react-native-community/slider/typings/index.d.ts
@@ -14,6 +14,14 @@ export interface SliderPropsAndroid extends ReactNative.ViewProps {
    * Color of the foreground switch grip.
    */
   thumbTintColor?: string;
+  /**
+   * modifies slider thickness. Android only.
+   */
+  sliderThickness?: number;
+  /**
+   * modifies slider corner roundness. Android only.
+   */
+  sliderCornerRoundness?: number;
 }
 
 export interface SliderRef {
