diff --git a/node_modules/@react-native-assets/slider/dist/components/Thumb.js b/node_modules/@react-native-assets/slider/dist/components/Thumb.js
index ff1d510..10a3746 100644
--- a/node_modules/@react-native-assets/slider/dist/components/Thumb.js
+++ b/node_modules/@react-native-assets/slider/dist/components/Thumb.js
@@ -66,7 +66,6 @@ function getThumbStyle(size, color) {
             backgroundColor: color,
             borderRadius: size / 2,
             overflow: 'hidden',
-            elevation: 10,
             zIndex: 10
         }
     }).thumb;
