diff --git a/node_modules/expo-wavy-slider/src/ExpoWavySliderModule.ts b/node_modules/expo-wavy-slider/src/ExpoWavySliderModule.ts
index cb4d400..473dceb 100644
--- a/node_modules/expo-wavy-slider/src/ExpoWavySliderModule.ts
+++ b/node_modules/expo-wavy-slider/src/ExpoWavySliderModule.ts
@@ -4,12 +4,14 @@ import { Platform } from 'react-native'
 
 type NativeSharedObject = InstanceType<typeof SharedObject>
 
-declare class ExpoWavySliderModule {
+class ExpoWavySliderModule {
+	// @ts-expect-error side effect for removing the declare kw
 	ObservableState: new (initial: { value: unknown }) => NativeSharedObject & {
 		getValue(): unknown
 		setValue(value: { value: unknown }): void
 		setOnChange(callback: object | null): void
 	}
+	// @ts-expect-error side effect for removing the declare kw
 	WorkletCallback: new (worklet: object) => NativeSharedObject
 }
 
diff --git a/node_modules/expo-wavy-slider/src/worklets.fx.ts b/node_modules/expo-wavy-slider/src/worklets.fx.ts
index 5579a47..176b1f0 100644
--- a/node_modules/expo-wavy-slider/src/worklets.fx.ts
+++ b/node_modules/expo-wavy-slider/src/worklets.fx.ts
@@ -85,5 +85,5 @@ try {
 		'[expo-wavy-slider] react-native-reanimated is required. Please install it first.',
 	)
 }
-installOnUIRuntime()
+installOnUIRuntime(worklets.getUIRuntimeHolder())
 registerSharedObjectSerializer()
