diff --git a/node_modules/expo-video/android/src/main/java/expo/modules/video/playbackService/ExpoVideoPlaybackService.kt b/node_modules/expo-video/android/src/main/java/expo/modules/video/playbackService/ExpoVideoPlaybackService.kt
index 54c0ad0..3430c6c 100644
--- a/node_modules/expo-video/android/src/main/java/expo/modules/video/playbackService/ExpoVideoPlaybackService.kt
+++ b/node_modules/expo-video/android/src/main/java/expo/modules/video/playbackService/ExpoVideoPlaybackService.kt
@@ -73,26 +73,6 @@ class ExpoVideoPlaybackService : MediaSessionService() {
   }
 
   fun registerPlayer(videoPlayer: VideoPlayer) {
-    appContext.mainQueue.launch {
-      val player = videoPlayer.player
-      if (mediaSessions[player] != null) {
-        return@launch
-      }
-
-      val mediaSession = MediaSession.Builder(this@ExpoVideoPlaybackService, player)
-        .setId("ExpoVideoPlaybackService_${player.hashCode()}")
-        .setCallback(VideoMediaSessionCallback())
-        .setCustomLayout(ImmutableList.of(seekBackwardButton, seekForwardButton))
-        .build()
-
-      // Replace the basic media session with a session connected to our playback service.
-      videoPlayer.mediaSession.release()
-      videoPlayer.mediaSession = mediaSession
-
-      mediaSessions[player] = mediaSession
-      addSession(mediaSession)
-      setShowNotification(videoPlayer.showNowPlayingNotification, player)
-    }
   }
 
   fun unregisterPlayer(player: ExoPlayer) {
diff --git a/node_modules/expo-video/android/src/main/java/expo/modules/video/player/VideoPlayer.kt b/node_modules/expo-video/android/src/main/java/expo/modules/video/player/VideoPlayer.kt
index ee4e976..cd1026d 100644
--- a/node_modules/expo-video/android/src/main/java/expo/modules/video/player/VideoPlayer.kt
+++ b/node_modules/expo-video/android/src/main/java/expo/modules/video/player/VideoPlayer.kt
@@ -88,7 +88,7 @@ class VideoPlayer(val context: Context, appContext: AppContext, source: VideoSou
 
   internal val firstFrameEventGenerator: FirstFrameEventGenerator
   val serviceConnection = PlaybackServiceConnection(WeakReference(this), appContext)
-  var mediaSession: MediaSession = buildBasicMediaSession(context, player)
+  //var mediaSession: MediaSession = buildBasicMediaSession(context, player)
   val intervalUpdateClock = IntervalUpdateClock(this)
 
   var playing by IgnoreSameSet(false) { new, old ->
@@ -374,7 +374,7 @@ class VideoPlayer(val context: Context, appContext: AppContext, source: VideoSou
       appContext?.reactContext?.unbindService(serviceConnection)
     }
     serviceConnection.playbackServiceBinder?.service?.unregisterPlayer(player)
-    mediaSession.release()
+    //mediaSession.release()
 
     VideoManager.unregisterVideoPlayer(this@VideoPlayer)
 
