diff --git a/node_modules/ffmpeg-kit-react-native/android/build.gradle b/node_modules/ffmpeg-kit-react-native/android/build.gradle
index 31004c8..56ac18f 100644
--- a/node_modules/ffmpeg-kit-react-native/android/build.gradle
+++ b/node_modules/ffmpeg-kit-react-native/android/build.gradle
@@ -127,5 +127,5 @@ dependencies {
   api 'com.facebook.react:react-native:+'
   // https://mvnrepository.com/artifact/com.arthenica/smart-exception-java
   implementation group: 'com.arthenica', name: 'smart-exception-java', version: '0.2.1'
-  api project(':ffmpeg-kit-binaries')
+  implementation files('../../../android/ffmpeg-kit-binaries/ffmpeg-kit-audio-6.0-2.aar')
 }