diff --git a/node_modules/react-native-paper/src/components/Menu/Menu.tsx b/node_modules/react-native-paper/src/components/Menu/Menu.tsx
index 9b53403..b352e68 100644
--- a/node_modules/react-native-paper/src/components/Menu/Menu.tsx
+++ b/node_modules/react-native-paper/src/components/Menu/Menu.tsx
@@ -346,6 +346,7 @@ const Menu = ({
 
     attachListeners();
     const { animation } = theme;
+    await new Promise(resolve => setTimeout(resolve, 1));
     Animated.parallel([
       Animated.timing(scaleAnimationRef.current, {
         toValue: { x: menuLayoutResult.width, y: menuLayoutResult.height },
@@ -365,11 +366,12 @@ const Menu = ({
     });
   }, [anchor, attachListeners, measureAnchorLayout, theme]);
 
-  const hide = React.useCallback(() => {
+  const hide = React.useCallback(async (callback?: () => unknown) => {
     removeListeners();
 
     const { animation } = theme;
     
+    await new Promise(resolve => setTimeout(resolve, 1));
     Animated.timing(opacityAnimationRef.current, {
       toValue: 0,
       duration: ANIMATION_DURATION * animation.scale,
@@ -380,6 +382,8 @@ const Menu = ({
       setRendered(false);
       prevRendered.current = false;
       focusFirstDOMNode(anchorRef.current);
+      scaleAnimationRef.current.setValue({x: 0, y: 0});
+      callback?.();
     });
   }, [removeListeners, theme]);
 
@@ -425,6 +429,7 @@ const Menu = ({
   }, [removeListeners, keyboardDidHide, keyboardDidShow]);
 
   React.useEffect(() => {
+    console.log('paper', visible, rendered);
     if (prevVisible.current !== visible) {
       prevVisible.current = visible;
 
@@ -636,7 +641,7 @@ const Menu = ({
           <Pressable
             accessibilityLabel={overlayAccessibilityLabel}
             accessibilityRole="button"
-            onPress={onDismiss}
+            onPress={() => hide(onDismiss)}
             style={styles.pressableOverlay}
           />
           <View
@@ -647,7 +652,7 @@ const Menu = ({
             accessibilityViewIsModal={visible}
             style={[styles.wrapper, positionStyle, style]}
             pointerEvents={pointerEvents}
-            onAccessibilityEscape={onDismiss}
+            onAccessibilityEscape={() => hide(onDismiss)}
             testID={`${testID}-view`}
           >
             <Animated.View
