diff --git a/node_modules/react-native-blob-util/android/build.gradle b/node_modules/react-native-blob-util/android/build.gradle
index 4fd9e0b..b7c1202 100644
--- a/node_modules/react-native-blob-util/android/build.gradle
+++ b/node_modules/react-native-blob-util/android/build.gradle
@@ -46,6 +46,7 @@ if (isNewArchitectureEnabled()) {
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', 30)
     buildToolsVersion safeExtGet('buildToolsVersion', '28.0.3')
+    namespace "com.ReactNativeBlobUtil"
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', 16)
         targetSdkVersion safeExtGet('targetSdkVersion', 30)
