diff --git a/node_modules/react-native-blob-util/fetch.js b/node_modules/react-native-blob-util/fetch.js
index 4f52ad3..3e9166f 100644
--- a/node_modules/react-native-blob-util/fetch.js
+++ b/node_modules/react-native-blob-util/fetch.js
@@ -271,7 +271,7 @@ export function fetch(...args: any): Promise {
                     if (options.session)
                         fs.session(options.session).add(data);
                 }
-                if ('uninit' in respInfo && respInfo.uninit) // event didn't fire yet so we override it here
+                if ('uninit' in respInfo && respInfo.uninit && responseInfo) // event didn't fire yet so we override it here
                     respInfo = responseInfo;
 
                 respInfo.rnfbEncode = rawType;
