diff --git a/node_modules/react-native-paper/src/components/ProgressBar.tsx b/node_modules/react-native-paper/src/components/ProgressBar.tsx
index dcdaf48..50ad810 100644
--- a/node_modules/react-native-paper/src/components/ProgressBar.tsx
+++ b/node_modules/react-native-paper/src/components/ProgressBar.tsx
@@ -203,7 +203,7 @@ const ProgressBar = ({
       accessibilityRole="progressbar"
       accessibilityState={{ busy: visible }}
       accessibilityValue={
-        indeterminate ? {} : { min: 0, max: 100, now: progress * 100 }
+        indeterminate ? {} : { min: 0, max: 100, now: Math.round(progress * 100) }
       }
       style={isWeb && styles.webContainer}
       testID={testID}
