diff --git a/node_modules/react-native-blob-jsi-helper/android/build.gradle b/node_modules/react-native-blob-jsi-helper/android/build.gradle
index 57945ae..95fe72a 100644
--- a/node_modules/react-native-blob-jsi-helper/android/build.gradle
+++ b/node_modules/react-native-blob-jsi-helper/android/build.gradle
@@ -45,6 +45,7 @@ repositories {
 
 android {
    compileSdkVersion safeExtGet("compileSdkVersion", 28)
+   namespace "com.reactnativeblobjsihelper"
     // Used to override the NDK path/version on internal CI or by allowing
    // users to customize the NDK path/version from their root project (e.g. for M1 support)
    if (rootProject.hasProperty("ndkPath")) {
@@ -73,10 +74,6 @@ android {
     }
   }
 
-  dexOptions {
-    javaMaxHeapSize "4g"
-  }
-
   compileOptions {
     sourceCompatibility JavaVersion.VERSION_1_8
     targetCompatibility JavaVersion.VERSION_1_8
diff --git a/node_modules/react-native-blob-jsi-helper/react-native-blob-jsi-helper.podspec b/node_modules/react-native-blob-jsi-helper/react-native-blob-jsi-helper.podspec
index 53ec31f..9ebd8f0 100644
--- a/node_modules/react-native-blob-jsi-helper/react-native-blob-jsi-helper.podspec
+++ b/node_modules/react-native-blob-jsi-helper/react-native-blob-jsi-helper.podspec
@@ -17,4 +17,8 @@ Pod::Spec.new do |s|
 
   s.dependency "React-Core"
   s.dependency "React"
+  
+  s.pod_target_xcconfig = {
+    "CLANG_CXX_LANGUAGE_STANDARD" => "c++17",
+  }
 end
