diff --git a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/AspectRatioFrameLayout.kt b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/AspectRatioFrameLayout.kt
index 834e5e0..25902fc 100644
--- a/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/AspectRatioFrameLayout.kt
+++ b/node_modules/react-native-video/android/src/main/java/com/brentvatne/exoplayer/AspectRatioFrameLayout.kt

@@ -72,12 +73,11 @@ class AspectRatioFrameLayout(context: Context) : FrameLayout(context) {
 
             ResizeMode.RESIZE_MODE_CENTER_CROP -> {
                 width = (measuredHeight * videoAspectRatio).toInt()
-
                 // Scale video if it doesn't fill the measuredWidth
                 if (width < measuredWidth) {
-                    val scaleFactor: Int = measuredWidth / width
-                    width *= scaleFactor
-                    height = measuredHeight * scaleFactor
+                    val scaleFactor: Float = measuredWidth.toFloat() / width
+                    width = (scaleFactor * width).toInt()
+                    height = (scaleFactor * height).toInt()
                 }
             }
 