diff --git a/node_modules/react-native-paper/src/components/Menu/Menu.tsx b/node_modules/react-native-paper/src/components/Menu/Menu.tsx
index db5e9a1..488a957 100644
--- a/node_modules/react-native-paper/src/components/Menu/Menu.tsx
+++ b/node_modules/react-native-paper/src/components/Menu/Menu.tsx
@@ -245,7 +245,7 @@ const Menu = ({
     if (visible) {
       onDismiss?.();
     }
-  }, [onDismiss, visible]);
+  }, [visible]);
 
   const handleKeypress = React.useCallback(
     (e: KeyboardEvent) => {
@@ -253,7 +253,7 @@ const Menu = ({
         onDismiss?.();
       }
     },
-    [onDismiss]
+    []
   );
 
   const removeListeners = React.useCallback(() => {
