diff --git a/node_modules/react-native-tab-view/src/TabBar.tsx b/node_modules/react-native-tab-view/src/TabBar.tsx
index e8d0b4c..8773976 100644
--- a/node_modules/react-native-tab-view/src/TabBar.tsx
+++ b/node_modules/react-native-tab-view/src/TabBar.tsx
@@ -448,7 +448,7 @@ export function TabBar<T extends Route>({
           {renderTabBarItem ? (
             renderTabBarItem(props)
           ) : (
-            <TabBarItem {...props} />
+            <TabBarItem {...props} key={props.key} />
           )}
         </>
       );
