diff --git a/node_modules/react-native-share-menu/android/build.gradle b/node_modules/react-native-share-menu/android/build.gradle
index cc6d464..da00e2e 100644
--- a/node_modules/react-native-share-menu/android/build.gradle
+++ b/node_modules/react-native-share-menu/android/build.gradle
@@ -13,7 +13,7 @@ apply plugin: 'com.android.library'
 
 android {
     compileSdkVersion safeExtGet('compileSdkVersion', rnsmDefaultCompileSdkVersion)
-
+    namespace "com.meedan"
     defaultConfig {
         minSdkVersion safeExtGet('minSdkVersion', rnsmDefaultCompileSdkVersion)
         targetSdkVersion safeExtGet('targetSdkVersion', rnsmDefaultCompileSdkVersion)