diff --git a/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js b/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js
index 2b6be5a..aa099b7 100644
--- a/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js
+++ b/node_modules/react-native-tab-view/lib/commonjs/PagerViewAdapter.js
@@ -54,7 +54,8 @@ function PagerViewAdapter({
     }
     if (indexRef.current !== index) {
       if (animationEnabled) {
-        pagerRef.current?.setPage(index);
+        pagerRef.current?.setPageWithoutAnimation(index);
+        position.setValue(index);
       } else {
         pagerRef.current?.setPageWithoutAnimation(index);
         position.setValue(index);