diff --git a/node_modules/react-native-paper/src/components/TouchableRipple/Pressable.tsx b/node_modules/react-native-paper/src/components/TouchableRipple/Pressable.tsx
index 446cb23..7bc8088 100644
--- a/node_modules/react-native-paper/src/components/TouchableRipple/Pressable.tsx
+++ b/node_modules/react-native-paper/src/components/TouchableRipple/Pressable.tsx
@@ -6,7 +6,7 @@ import type {
   View,
   ViewStyle,
 } from 'react-native';
-import { Pressable as PressableNative } from 'react-native';
+import { Pressable as PressableNative } from 'react-native-gesture-handler';
 
 // This component is added to support type-safe hover and focus states on web
 // https://necolas.github.io/react-native-web/docs/pressable/
