diff --git a/node_modules/react-native-reanimated/src/shareables.ts b/node_modules/react-native-reanimated/src/shareables.ts
index 44e961f..11bf641 100644
--- a/node_modules/react-native-reanimated/src/shareables.ts
+++ b/node_modules/react-native-reanimated/src/shareables.ts
@@ -343,11 +343,7 @@ function freezeObjectIfDev<T extends object>(value: T) {
         return element;
       },
       set() {
-        logger.warn(
-          `Tried to modify key \`${key}\` of an object which has been already passed to a worklet. See 
-https://docs.swmansion.com/react-native-reanimated/docs/guides/troubleshooting#tried-to-modify-key-of-an-object-which-has-been-converted-to-a-shareable 
-for more details.`
-        );
+        
       },
     });
   });
