diff --git a/node_modules/react-native-blob-util/android/src/main/AndroidManifest.xml b/node_modules/react-native-blob-util/android/src/main/AndroidManifest.xml
index 45111e4..114dad6 100644
--- a/node_modules/react-native-blob-util/android/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-blob-util/android/src/main/AndroidManifest.xml
@@ -1,4 +1,5 @@
-<manifest xmlns:android="http://schemas.android.com/apk/res/android">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+          package="com.ReactNativeBlobUtil">
 
     <!-- Required to download files from Google Play -->
     <uses-permission android:name="android.permission.INTERNET" />
diff --git a/node_modules/react-native-blob-util/fetch.js b/node_modules/react-native-blob-util/fetch.js
index 4f52ad3..3e9166f 100644
--- a/node_modules/react-native-blob-util/fetch.js
+++ b/node_modules/react-native-blob-util/fetch.js
@@ -271,7 +271,7 @@ export function fetch(...args: any): Promise {
                     if (options.session)
                         fs.session(options.session).add(data);
                 }
-                if ('uninit' in respInfo && respInfo.uninit) // event didn't fire yet so we override it here
+                if ('uninit' in respInfo && respInfo.uninit && responseInfo) // event didn't fire yet so we override it here
                     respInfo = responseInfo;
 
                 respInfo.rnfbEncode = rawType;
